README.md

```diff
@@ -231,172 +259,159 @@ curl -s -i https://api.travis-ci.org/auth/github \
 
-See the [Script structure](#script-structure) section below for details on the `script.sh` format.
+### Script structure
 
-#### Script structure
-
-This `script.sh` is structured like:
+This `script.sh` in each package is structured like:
 
 ```bash
 #!/usr/bin/env bash
 
+# This is required for every package.
 MASON_NAME=libuv
 MASON_VERSION=0.11.29
-MASON_LIB_FILE=lib/libuv.a
-MASON_PKGCONFIG_FILE=lib/pkgconfig/libuv.pc
-```
 
-Declare these variables first. `MASON_NAME` and `MASON_VERSION` are mandatory. If the install script build a static library, specify the relative path in the installation directory in `MASON_LIB_FILE`. This is used to check whether an installation actually exists before proceeding to download/build the library anew. You can optionally specify `MASON_PKGCONFIG_FILE` as the relative path to the pig-config file if the library has one. If the library doesn't have one, you need to override the functions `mason_cflags` and `mason_ldflags` (see below).
+# This is required if Mason will need to build a static library. Specify the relative path in the
+# installation directory.
+MASON_LIB_FILE=lib/libuv.a
 
-Then, we're loading the build system with
+# You can specify the relative path to the pkg-config file if Mason needs to build your code before
+# packaging. If the library doesn't have a pkg-config file, you will need to override the functions
+# `mason_cflags` and `mason_ldflags`.
+MASON_PKGCONFIG_FILE=lib/pkgconfig/libuv.pc
 
-```bash
+# This is required when you need to load the build system to build your code before packaging. You
+# con't need this line if you are packaging header-only code.
 . ${MASON_DIR}/mason.sh
-```
 
-Next, we're defining a function that obtains the source code and unzips it:
-
-```bash
+# Overriding this Mason function is required for all pakcages so Mason knows where to obtain your
+# source code.
 function mason_load_source {
     mason_download \
         https://github.com/joyent/libuv/archive/v0.11.29.tar.gz \
         5bf49a8652f680557cbaf335a160187b2da3bf7f
 
+    # This unpacks the archive into the `mason_packages/.build` folder. If the tarball is BZip2
+    # compressed, you can also use `mason_extract_tar_bz2` instead.
     mason_extract_tar_gz
 
-    export MASON_BUILD_PATH=${MASON_ROOT}/.build/libuv-${MASON_VERSION}
+    # This variable contains the path to the unpacked folder inside the `.build` directory.
+    export MASON_BUILD_PATH=${MASON_ROOT}/.build/${MASON_NAME}-${MASON_VERSION}
 }
-```
-
-In that function, you should use `mason_download` as a shortcut to download the tarball. The second argument to is a hash generated with `git hash-object` and used to verify that the source code downloaded matches the expected file. The function also caches downloaded tarballs in the `mason_packages/.cache` folder. You can get the hash object by running `./mason build <package> <version>` locally and this should fail with a message of the expected hash.
```

Comment:
Worth it to retain
> The function also caches downloaded tarballs in the `mason_packages/.cache` folder.

here still? The only other place cache is mentioned is at the bottom of the doc.

Reply:
@GretaCB - I removed it from here because it wasn't helping my understanding of how we git the checksum back and I wanted to reduce clutter in the **Getting started** section. I could add it to the cache section at the bottom and list everything we store there. What do you think? Did it actually help your understanding of the process to create a package?

Reply:
> Did it actually help your understanding of the process to create a package?

Great question. My assumption is that this [`Script structure`](https://github.com/mapbox/mason/blob/0800a52406e28955c3ee3204a35ca3c9a64302d6/README.md#script-structure) section is the main place to go for details on what each function does. And part of what `mason_load_source` does is add to `/.cache`. I can see it being handy to have that in one place for reference.

What are your thoughts on `Script Struture` being a standalone section, but _referenced_ for Getting started?

Reply:
I agree. I'll add it to the `Script structure` section.

I think moving `Script structure` out of the `Creating a package` section is good idea too. Also: `Mason variables` and `Mason functions` can be moved up a level as well. 👍